### PR TITLE
Move add-to-cart, group-menu and conflict related UI components to new library

### LINKED ISSUE #8414: Move addToCart, group menu and conflict related UI components to new library

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts]: I believe we are subscribing twice in the template to the `routerData$`:
- the first subscription is when we're subscribing directly to it
- the second subscription happens indirectly, when subscribing to `configuration$`

We could minimize this by having only one stream, for example:

```ts
  anAppropriateName$: Observable<{
    routerData: ConfigurationRouter.Data;
    configuration: Configurator.Configuration;
  }> = this.configRouterExtractorService
    .extractRouterData()
    .pipe(
      switchMap((routerData) =>
        this.configuratorCommonsService
          .getConfiguration(routerData.owner)
          .pipe(map((configuration) => ({ routerData, configuration })))
      )
    );
```

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts]: Instead of hardcoding the URL, I recommend to just pass the route name and a parameter. 
Customers can extend and change the route structure, and this will not work in that case.

For an example, please take a look at other usages of this `go` method, e.g.:

```ts
  this.routingService.go({
          cxRoute: 'product',
          params: { code: this.defaultPreviewProductCode, name: '' },
        });
```

File: feature-libs/product/configurators/common/src/components/config-conflict-description/config-conflict-description.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-conflict-description/config-conflict-description.component.ts]: Let's follow the convention and use a lowercase `g` here:
```ts
groupType = Configurator.GroupType
```

File: feature-libs/product/configurators/common/src/components/config-conflict-suggestion/config-conflict-suggestion.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-conflict-suggestion/config-conflict-suggestion.component.ts]: Let's do the same here - use the lowercase `g`.
Please check other usages as well, and change if necessary.

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Please follow the same approach as done here https://github.com/SAP/spartacus/pull/8193/files#diff-1731239c81ecd8c50e660816f5bf6c0fR34 - 
- use this method: https://github.com/SAP/spartacus/pull/8193/files#diff-1731239c81ecd8c50e660816f5bf6c0fR34
- it wraps the logic in a check if the platform is browser or a server. This will not break if the app is being server-side rendered (SSR)

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: I'm curious why was this necessary?

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Let's follow the convention, and use camil case for variable names.

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Let's switch these to `protected`. All injected dependencies should always be `protected` for extensibility purposes.

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Is there a type that we can use for `event`?

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Is there a type for this `event` that we can use?

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts:36]: Could you explain to me why the type gets both ConfigurationRouter.Data and Configurator.Configuration
Observable<{
    routerData: ConfigurationRouter.Data;
    configuration: Configurator.Configuration;
  }>
and not simply Observable<Configurator.Configuration>

File: feature-libs/product/configurators/common/src/components/config-conflict-description/config-conflict-description.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-conflict-description/config-conflict-description.component.ts]: Done!

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: This method will be removed.
The PR is waiting for approval:  https://github.com/SAP/spartacus/pull/8539/files#diff-aba2ad5c841b5a5d9532088d1c4486f0
As soon as this PR is merged. I will fix it correspondingly. The other PR has a dependency to this one.

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Done

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Done

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: is **KeyboardEvent** or better **any** as a type okay?

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: is **KeyboardEvent** or better **any** as a type okay?

File: feature-libs/product/configurators/common/src/components/config-conflict-suggestion/config-conflict-suggestion.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-conflict-suggestion/config-conflict-suggestion.component.ts]: Done!

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: I will come back to you with the answer to your question

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts:36]: At this point we simply need configuration$ from type Observable<Configurator.Configuration>

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: We had an issue by back navigation fromthe overview page to the configuration: https://github.com/SAP/spartacus/issues/7418

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts]: Done

File: feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts]: Nice idee but it is not clear for me how to get a certain part from the observable in HTML with async it does not work

Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-add-to-cart-button/config-add-to-cart-button.component.ts]: Done!

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: The previously mentioned PR has been integrated. I removed **scrollToVariantConfigurationHeader**. The scrolling to the expected element happens by means of the configUtilService.

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Please always use a specific type, rather than `any`.
If the `event` can have multiple types, you can specify them like this; `KeyboardEvent | MouseEvent | etc.`

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Please see https://github.com/SAP/spartacus/pull/8434#discussion_r469107010

File: feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts]: Thanks. At the moment we are expecting only a keyboard event. If we expect one more event later, we will add it, as you mentioned above, via pipe operator.

**Review by znikola — APPROVED** [review]

**Comment by Paras5393** [comment]
@ChristophHi Was there any resolution for the above mentioned issue? 
I have been facing this for past couple of days but still remains unresolved. Kindly point in the right direction if possible. 

Regards.

**Comment by ChristophHi** [comment]
What issue are you talking about exactly? This is a PR that was used in the development process of the product configurator feature lib, it did not address specific issues

**Pull request merged**
